src/checkable_continuous_flyout.js

```diff
@@ -0,0 +1,209 @@
+import * as Blockly from 'blockly';
+import {ContinuousFlyout} from '@blockly/continuous-toolbox';
+
+export class CheckableContinuousFlyout extends ContinuousFlyout {
+  /**
+   * Size of a checkbox next to a variable reporter.
+   * @type {number}
+   * @const
+   */
+  static CHECKBOX_SIZE = 25;
+
+  /**
+   * Amount of touchable padding around reporter checkboxes.
+   * @type {number}
+   * @const
+   */
+  static CHECKBOX_TOUCH_PADDING = 12;
+
+  /**
+   * SVG path data for checkmark in checkbox.
+   * @type {string}
+   * @const
+   */
+  static CHECKMARK_PATH =
+      'M' + CheckableContinuousFlyout.CHECKBOX_SIZE / 4 +
+      ' ' + CheckableContinuousFlyout.CHECKBOX_SIZE / 2 +
+      'L' + 5 * CheckableContinuousFlyout.CHECKBOX_SIZE / 12 +
+      ' ' + 2 * CheckableContinuousFlyout.CHECKBOX_SIZE / 3 +
+      'L' + 3 * CheckableContinuousFlyout.CHECKBOX_SIZE / 4 +
+      ' ' + CheckableContinuousFlyout.CHECKBOX_SIZE / 3;
+
+  /**
+   * Size of the checkbox corner radius
+   * @type {number}
+   * @const
+   */
+  static CHECKBOX_CORNER_RADIUS = 5;
+
+  /**
+   * @type {number}
+   * @const
+   */
+  static CHECKBOX_MARGIN = ContinuousFlyout.prototype.MARGIN;
+
+  /**
+   * Total additional width of a row that contains a checkbox.
+   * @type {number}
+   * @const
+   */
+  static CHECKBOX_SPACE_X =
+      CheckableContinuousFlyout.CHECKBOX_SIZE +
+      2 * CheckableContinuousFlyout.CHECKBOX_MARGIN;
+
+
+  constructor(workspaceOptions) {
+    super(workspaceOptions);
+    CheckableContinuousFlyout.CHECKBOX_MARGIN = this.MARGIN;
+
+    /**
+     * Map of checkboxes that correspond to monitored blocks.
+     * Each element is an object containing the SVG for the checkbox, a boolean
+     * for its checked state, and the block the checkbox is associated with.
+     * @type {!Object.<string, !Object>}
+     * @private
+     */
+    this.checkboxes_ = {};
+  }
+
+  show(flyoutDef) {
+    this.clearOldCheckboxes();
+    super.show(flyoutDef);
+  }
+
+  clearOldCheckboxes() {
+    for (const checkboxId in this.checkboxes_) {
+      if (!this.checkboxes_.hasOwnProperty(checkboxId)) continue;
+      const checkbox = this.checkboxes_[checkboxId];
+      checkbox.block.flyoutCheckbox = null;
+      checkbox.svgRoot.remove();
+    }
+    this.checkboxes_ = {};
+  }
+
+  addBlockListeners_(root, block, rect) {
```

Comment:
This is the sketchy part I think should be moved to use the [proposed API](https://docs.google.com/document/d/1Ac_3OzVWuF5rWRn9F3tD12KqTFhdaMT0H1OHV7aE_9s/edit?usp=sharing&resourcekey=0-NV1y4ASQ3kAj5K6WTeOTyg) once it's implemented. Cuz the [original implementation](https://github.com/google/blockly/blob/e2df0fc2885cb994886c0af863035f50bb849948/core/flyout_base.ts#L1002) is actually just about adding listeners.

Reply:
Yep, filed #44 to track.